Cargo.toml

```diff
@@ -14,18 +14,25 @@ description = "Fuel cryptographic primitives."
 borrown = "0.1"
 fuel-types = { version = "0.5", default-features = false }
 lazy_static = { version = "1.4", optional = true }
+serde = { version = "1.0", default-features = false, features = ["derive"], optional = true }
+sha2 = { version = "0.9", default-features = false }
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+# the deterministic arguments of the signature (key, nonce, message), as defined in the RFC-6979
+rand = { version = "0.8", default-features = false, features = ["alloc"], optional = true }
```

Comment:
We are requiring `alloc` but it isn't used in the crate. We need to create a new feature `alloc` and enable this behavior.

Reply:
This is the main thing to add